tools/TestImpact/Test-CertifiedShardMap.ps1

```
@@ -0,0 +1,120 @@
+    $mapSha = [string] $Map.sha
+    $candidateSha = [string] $Certificate.candidateMapSha
+    if ($mapSha -notmatch '^[0-9a-f]{40}$') { throw 'map sha is not a full lowercase commit id' }
+    if ($candidateSha -cne $mapSha) { throw 'certificate and map identify different source commits' }
+    if ([string] $Certificate.auditSourceSha -notmatch '^[0-9a-f]{40}$') {
+        throw 'auditSourceSha is not a full lowercase commit id'
+    }
+}
+
+if ($SelfTest) {
+    $failures = [System.Collections.Generic.List[string]]::new()
+    function Assert-Throws {
```

PSScriptAnalyzer (1.25.0)

[warning] The cmdlet 'Assert-Throws' uses a plural noun. A singular noun should be used instead.

Suggested fix: Singularized correction of 'Assert-Throws'

(PSUseSingularNouns)